internal/pkg/convert/node.go

```diff
@@ -311,7 +319,19 @@ func (node *NodeLLB) stepScripts(root llb.State, i int, step v1alpha2.Step) llb.
 				runOptions = append(runOptions, llb.IgnoreCache)
 			}
 
-			root = root.Run(runOptions...).Root()
+			scriptRoot = scriptRoot.Run(runOptions...).Root()
+		}
+
+		if script.Detached {
+			scriptRoot = scriptRoot.File(
+				llb.Mkdir("/empty", constants.DefaultDirMode),
+			)
+
+			root = root.File(
+				llb.Copy(scriptRoot, "/empty", "/", defaultCopyOptions(node.Graph.Options, false)),
```

Comment:
Couldn't comprehend this. Do we just overwrite the result with empty dir?

Reply:
no... that's a bit of a hack, I asked Tonis a long time ago about a better way, but there's nothing.

`/empty` is empty in the source, and we copy it to `/`, so it actually copies nothing, but this creates a dependency between LLB stages, otherwise buildkit won't even execute `test` script

Reply:
gotcha